packages/graphql/src/translate/where/property-operations/create-connection-operation.ts

```diff
@@ -98,3 +99,79 @@ export function createConnectionOperation({
 
     return CypherBuilder.and(...operations) as CypherBuilder.BooleanOp | undefined;
 }
+
+function createConnectionWherePropertyOperation({
+    context,
+    whereInput,
+    relationshipRef,
+    targetNode,
+    node,
+    edge,
+}: {
+    whereInput: ConnectionWhereArg;
+    context: Context;
+    node: Node;
+    edge: Relationship;
+    relationshipRef: CypherBuilder.Relationship;
+    targetNode: CypherBuilder.Node;
+}): CypherBuilder.ComparisonOp | CypherBuilder.BooleanOp | CypherBuilder.RawCypher | CypherBuilder.Exists | undefined {
+    const params = Object.entries(whereInput).map(([key, value]) => {
+        if (key === "AND" || key === "OR") {
+            const subOperations = (value as Array<any>).map((input) => {
+                return createConnectionWherePropertyOperation({
+                    context,
+                    whereInput: input,
+                    relationshipRef,
+                    targetNode,
+                    node,
+                    edge,
+                });
+            });
+            if (key === "AND") {
+                return CypherBuilder.and(...filterTruthy(subOperations));
+            }
+            if (key === "OR") {
+                return CypherBuilder.or(...filterTruthy(subOperations));
+            }
+        }
+
+        if (key.startsWith("edge")) {
+            const nestedProperties: Record<string, any> = value;
+            const result = createCypherWhereParams({
+                targetElement: relationshipRef,
+                whereInput: nestedProperties,
+                context,
+                element: edge,
+            });
+
+            return result;
+        }
+
+        if (key.startsWith("node") || key.startsWith(node.name)) {
+            // TODO: improve
```

Comment:
This `TODO` doesn't really offer any value right now - worth adding some context?

Reply:
```suggestion
            // TODO: improve nodeOn properties generation
```